### PR TITLE
Feature survey

### PR DESCRIPTION
**Pull request by techaffinity**
Surveys has been implemented in the backend as Wagtail page and models have been created to allow admin to add surveys(questions) and can map to any of the pages. We have also developed front end pages to allow users to view surveys in the front end pages and can answer to those questions.

We are facing the issue with front end while user adding answers to those questions. Answers are not getting created for those questions and the error is related to depth column in Wagtail Core.

Please find below for your reference,

File Location: Surveys/views.py - Line no:18

Exception Type: IntegrityError
Exception Value:     wagtailcore_page.depth may not be NULL

We are checking this in our end and could update on this once we able to fix this issue and make sure the users can add answers to the questions.

### CODE REVIEW

**Comment by smn** [comment]
The build is failing due to issues unrelated to the `Exception Type: IntegrityError
Exception Value: wagtailcore_page.depth may not be NULL` error. Please fix the build so we at least eliminate any errors originating from that.

**Comment by techaffinity** [comment]
We have checked this in our end and tried running the test [ python manage.py test -v1 ]

We could not see this migration error in our end and could run the test successfully. We have even tried mapping to a new database and applied migrations. We could not see this migration error. We will try to reproduce it if we could get that. Please advice.

Error:
django.db.migrations.graph.NodeNotFoundError: Migration surveys.0001_initial dependencies reference nonexistent parent node (u'wagtailcore', u'0017_auto_20150731_1102')

We have solved one of the flake8 standard issue and updated the build.

**Comment by smn** [comment]
You were referring to a migration that simply doesn't exist in Wagtail. If you look at the list of available migrations at https://github.com/torchbox/wagtail/tree/master/wagtail/wagtailcore/migrations you'll notice that `0017_auto_20150731_1102.py` isn't available there. I think perhaps you locally made some changes that somehow ended up as a wagtail migration which are now available on your system but aren't in the official wagtail distribution.

Instead I've referred to the last available wagtail migration: `0016_change_page_url_path_to_text_field`.

I've also merged in the changes made in `develop` and resolved the conflicts there.

File: ndohyep/polls/detail.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+	<form action="{% url 'vote' question.id %}" method="post">
+		{% csrf_token %}
+		{% for choice in question.choice_set.all %}
+		    <input type="radio" name="choice" id="choice{{ forloop.counter }}" value="{{ choice.id }}" />
+		    <label for="choice{{ forloop.counter }}">{{ choice.choice_text }}</label><br />
+		{% endfor %}
+		<input type="submit" value="Vote" />
+	</form>
+	<a href="{% url 'results' question.id %}">See Results</a>
```
Comment (smn) [inline ndohyep/polls/detail.html]: `{% trans ... %}`

File: ndohyep/polls/detail.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{% endblock %}
```
Comment (smn) [inline ndohyep/polls/detail.html]: Missing new line.

File: ndohyep/polls/index.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{% load i18n %}
+{% if latest_question_list %}
+    <ul>
+    {% for question in latest_question_list %}
+        <li><a href="/polls/{{ question.id }}/">{{ question.question_text }}</a></li>
```
Comment (smn) [inline ndohyep/polls/index.html]: `{% url %}`

File: ndohyep/polls/index.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    {% endfor %}
+    </ul>
+{% else %}
+    <p>No polls are available.</p>
```
Comment (smn) [inline ndohyep/polls/index.html]: `{% trans %}`

File: ndohyep/polls/index.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+{% endif %}
```
Comment (smn) [inline ndohyep/polls/index.html]: Missing new line.

**Comment by smn** [comment]
I've had a look at the pull-request. This is now the third I'm reviewing and I'm flagging the same issues around the use of `{% trans ... %}` and `{% url ... %}`, missing new lines etc. Doing these small things well matters to us and by now you know well what I'm looking for first in a review. Please attend to those first and let me know when done so I can do a review of the actual features.

**Comment by smn** [comment]
I've merged #5 into `develop` and `develop` back into this pull-request.

**Comment by techaffinity** [comment]
Simon we have already checked these items and resolved in the feature questions branch. Since we updated these in feature questions branch, the code we are looking here is in surveys feature which has yet to be merged to get the latest update from questions branch. We could see that you have merged now and get the latest code.

**Comment by smn** [comment]
:+1: please go ahead and merge.

**Pull request merged**